cmd/prometheus/main_test.go

```diff
@@ -798,3 +807,73 @@ runtime:
 		})
 	}
 }
+
+// TestHeadCompactionWhileScraping verifies that running a head compaction
+// concurrently with a scrape does not trigger the data race described in
+// https://github.com/prometheus/prometheus/issues/16490.
+func TestHeadCompactionWhileScraping(t *testing.T) {
+	t.Parallel()
+
+	tmpDir := t.TempDir()
+	configFile := filepath.Join(tmpDir, "prometheus.yml")
+
+	port := testutil.RandomUnprivilegedPort(t)
+	config := fmt.Sprintf(`
+global:
+  scrape_timeout: 200ms
+scrape_configs:
+ - job_name: 'self1'
```

Comment:
only exercising the text parser
(prometheus' /metrics doesn't support OM, yet?)